# TODO remove usage of onyx_backend_main.py
diff --git a/backend/onyx/main.py b/backend/onyx/main.py
index 1929a6666..90218921b 100644
--- a/backend/onyx/main.py
+++ b/backend/onyx/main.py
@@ -137,6 +137,8 @@ from shared_configs.configs import POSTGRES_DEFAULT_SCHEMA
 from shared_configs.configs import SENTRY_DSN
 from shared_configs.contextvars import CURRENT_TENANT_ID_CONTEXTVAR
 
+from onyx.httpx.custom_cert_oauth import CustomOpenID
+
 logger = setup_logger()
 
 file_handlers = [
@@ -470,7 +472,7 @@ def get_application(lifespan_override: Lifespan | None = None) -> FastAPI:
         include_auth_router_with_prefix(
             application,
             create_onyx_oauth_router(
-                OpenID(
+                CustomOpenID(
                     OAUTH_CLIENT_ID,
                     OAUTH_CLIENT_SECRET,
                     OPENID_CONFIG_URL,
